Undo remove openmp from Axom section in SetupThirdParty

This line can be removed in an upcoming Axom update:

```cmake
list(REMOVE_ITEM _axom_exported_targets openmp)
```

See https://github.com/LLNL/Tribol/pull/52/files#diff-7c7bae65f535f61f97e00bb99c48133673c6d8039edb9d526d51cc45c03cfca6R36

              Also, i had to remove openmp from list, since it was included in another export set

From serac codevelop: `CMake Error: install(EXPORT "tribol-targets" ...) includes target "openmp" more than once in the export set.`

_Originally posted by @chapman39 in https://github.com/LLNL/Tribol/pull/52#discussion_r1454219835_

--- a/cmake/SetupThirdParty.cmake
+++ b/cmake/SetupThirdParty.cmake
@@ -79,7 +79,6 @@ if (TARGET axom)
     # Add components and dependencies of axom to this export set but don't prefix it with tribol::
     # NOTE(chapman39@llnl.gov): Cannot simply install axom, since it is an imported library
     set(_axom_exported_targets ${axom_exported_targets})
-    list(REMOVE_ITEM _axom_exported_targets openmp)
     install(TARGETS              ${_axom_exported_targets}
             EXPORT               tribol-targets
             DESTINATION          lib)

